equinix/resource_fabric_connection_acc_test.go

```diff
@@ -0,0 +1,998 @@
+				),
+				ExpectNonEmptyPlan: true,
+			},
+			{
+				Config: testAccFabricCreateEPLConnectionConfig(100),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(
+						"equinix_fabric_connection.test", "name", fmt.Sprint("fabric_tf_acc_test_CCEPL")),
+					resource.TestCheckResourceAttr(
+						"equinix_fabric_connection.test", "bandwidth", fmt.Sprint("100")),
+				),
+				ExpectNonEmptyPlan: true,
+			},
+		},
+	})
+}
+
+func checkConnectionDelete(s *terraform.State) error {
+	client := testAccProvider.Meta().(*Config).fabricClient
+	ctx := context.Background()
+	ctx = context.WithValue(ctx, v4.ContextAccessToken, testAccProvider.Meta().(*Config).FabricAuthToken)
+	//conn := v4.Connection{}
+	//var err error
+	//counter := 0
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "equinix_fabric_connection" {
+			continue
+		}
+
+		//TODO Use retry terraform helpers - Move to delete Func
+		_, err := waitUntilConnectionDeprovisioned(rs.Primary.ID, client, ctx)
+		if err != nil {
+			return fmt.Errorf("API call failed while waiting for resource deletion")
+		}
+		/*for conn.State == nil || "DEPROVISIONED" != *conn.State {
+			time.Sleep(30 * time.Second)
+			conn, _, err = client.ConnectionsApi.GetConnectionByUuid(ctx, rs.Primary.ID, nil)
+			if err != nil {
+				return fmt.Errorf("API call failed for the resource ")
+			}
+			if counter >= 4 {
+				break
+			}
+			counter++
+		}*/
+	}
+	return nil
+}
+
+func waitUntilConnectionDeprovisioned(uuid string, client *v4.APIClient, ctx context.Context) (v4.Connection, error) {
+	log.Println("Waiting for connection to be in active state ")
+	stateConf := &resource.StateChangeConf{
+		Target: []string{"DEPROVISIONED"},
+		Refresh: func() (interface{}, string, error) {
+			dbConn, _, err := client.ConnectionsApi.GetConnectionByUuid(ctx, uuid, nil)
+
+			if err != nil {
+				return "", "", err
+			}
+			updatableState := ""
+			if "DEPROVISIONED" == *dbConn.State {
+				updatableState = string(*dbConn.State)
+			}
+			return dbConn, updatableState, nil
+		},
+		Timeout:    3 * time.Minute,
+		Delay:      30 * time.Second,
+		MinTimeout: 30 * time.Second,
+	}
+
+	inter, err := stateConf.WaitForStateContext(ctx)
+	dbConn := v4.Connection{}
+
+	if err == nil {
+		dbConn = inter.(v4.Connection)
+	}
+	return dbConn, err
+}
+
+func testAccFabricCreateEPLConnectionConfig(bandwidth int32) string {
+
+	return fmt.Sprintf(`resource "equinix_fabric_connection" "test" {
+		type = "EVPL_VC"
+		name = "fabric_tf_acc_test_CCEPL"
+		notifications{
+			type = "ALL"
+			emails = ["test@equinix.com","test1@equinix.com"]
+		}
+		order {
+		purchase_order_number = "1-129105284100"
+			}
+		bandwidth = %d
+		a_side {
+		access_point {
+			type = "COLO"
+				port {
+				uuid = "c4d9350e-77cd-7cdd-1ce0-306a5c00a600"
+				}
+				link_protocol {
+					type= "QINQ"
+					vlan_s_tag= 1231
+				}
+			}
+		}
+		z_side {
+		access_point {
```

Comment:
there is a linter than could help clean up the formatting in these tests. I can't find the project at the moment.

I'm not worried about it, but it would make these easier to read.